### ISSUE TITLE
Change nozzle temperature for layer above support interface

### ISSUE THREAD

**Issue by movinimage** [SoftFever/OrcaSlicer#7647]
### Is there an existing issue for this feature request?

- [x] I have searched the existing issues

### Is your feature request related to a problem?

I am always looking for ways to have the cleanest supported surfaces - without the tool/filament change required to use a dissimilar support filament. I just saw a demo in which the extrusion temp was lowered for the supported parts of the the first layer above the support interface. The author of the video was able to achieve a 0z-interface gap, and still easily peel the support off.

The author of the video was showing off his inductive hotend which makes rapid heating/cooling possible. But, I think this could be done without that. It might require a retract, lift, wait for nozzle temp change, prime, continue. It might not have to be that complicated though - I'm not thinking outside the box this morning.

Here is the video, queued up to the exact place where he shows of the 0z-speration supports achieved through lowering the nozzle temp: 

  [![YouTube Video](https://img.youtube.com/vi/XphpaHd8Q9s/0.jpg)](https://youtu.be/XphpaHd8Q9s?si=ZviMYGvCbHiT0DfJ&t=293)

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

-Turn on the feature
-Set a temp amount to lower nozzle
-Offer the option to put in a 'pause' to reach the required temp (which may require a retract and movement away) - this would be useful when a layer contains a supported area and a non-supported area that needs to be strong (two temps on one layer). When unchecked, the printer would simply make best efforts to reach the temperature differential, but would not stop printing - this might work when an nearly entire layer requires support, or strength isn't that much of a concern.

Maybe two checkboxes and a field in support settings:

Checkbox: Lower nozzle temp for supported areas of first layer above support? [x]
Field: Lower nozzle temp by deg C [  15  ]
Checkbox: Pause to wait for nozzle temp when moving from supported to unsupported (or back) [x]

### Describe alternatives you've considered

Hand modifying nozzle temp in gcode for layers or parts of layers

**Comment by ShaneDelmore** [comment]
This is a great idea.  My current workaround is to set my fan to 100% for support interface and set the print speed to 10mm/s, this gives a long time for the support interface to cool down, but in the process also lets the rest of the model be over-cooled and results in very easy to remove supports, great surface quality, but poor layer adhesion for the whole model for that layer.  This solution would be a great improvement.

**Comment by igiannakas** [comment]
I saw that video yesterday too. While its "easy" to implement even without any code changes today (orca exposes the extrusion roles already in the gcode editor), the neater trick would be to get this to work with existing hot ends! 

Basically what I am thinking is, switch off the heater ahead of time, to allow the hot end to cool down before printing the supported area while material is still being extruded for the internal sparse infill. This would make it less reliant on absolute heater speed.

Alternatively, if the printer is equipped with a nozzle parking area, use it to park the hot end to cool it down and warm it up.

Basically there are ways to do this today, without any expensive equipment as long as we are "clever" in how we handle temperature setting ahead of time and if the printer is equipped with hardware to stop the nozzle from leaking while heating up/cooling down.

eg:
![IMG_3750](https://github.com/user-attachments/assets/b095e11b-e6ba-41d3-865c-c860f021f933)
![IMG_3749](https://github.com/user-attachments/assets/d36236fc-939d-46f4-979a-bc82a7a634dd)

**Comment by movinimage** [comment]
Good ideas...thats the part I wasn't sure how to implement - what to do with the oozing hotend while we wait for the temp to drop...and then what to do when we want to get back to temp after.  Dropping the temp is much faster because the flowing filament pulls a lot of heat away - so the infill idea could work for that.

You could also do the temp change on a prime tower.. Retract, travel to prime tower, extrude until lower temp reached, go back and continue...And the same for raising the temp. This would be clean and would work even if there wan't enough infill space to do the temp change....I do hate prime towers though! :)

**Comment by officialpoiuytrewq4645** [comment]
This should totally be a thing if only i had any idea where to even start with this codebase (is there some kind of implementation guide around?)
Filament could just be pooped into infill areas, depending on geometry

**Comment by coryrc** [comment]
I think the proposal isn't fully baked. Consider this cross section:

```
-------
|  X  |
```

The `X` is a support, `|` are walls. There's bridging on either side of the support to the walls. Normally it's printed with a single stroke across the entire width. If you have to move and wait to cool down, you couldn't print this: the sides would be cantilevered instead of bridged.
